src/OpenXR.cpp

```diff
@@ -618,6 +796,52 @@ bool OpenXR::update_vr_poses()
     return true;
 }
 
+void OpenXR::update_poll_hand_poses()
```

Comment:
I think I understand the code otherwise, but I have some questions about this function:

- Is xrPollEvent needed. If it is, why? There's nothing that's done with the event data here as far as I can see
- You say the rest of the code is probably not needed. I agree that the for loop at least is useless as only thing it does is that it sets `handActive[hand] = false` which is always false, isn't it? Or is something else writing into `handActive` array that's not obvious from here. I would guess `xrSyncActions` is needed, maybe?

Reply:
`xrPollEvent` is needed to get it running. But perhaps this should be in the main loop to be able to detect state changes if headset-connection or something changes (reference space, interaction profile). If I recall correctly openRBRVR just polls in the startup to search for one specific message and then never polls again. 

Currently it's not needed for regular openRBRVR, so I put it in `update_poll_hand_poses()` to isolate it and not change how the openRBRVR event loop works currently.

the //TODO: is incorrect, `xrSyncActions` is responsible for updating/sending the pose and is needed.

I tried commenting out the rest of the code with `xrGetActionStatePose` and  `xrLocateSpace`, and it stopped working.

The documentation is very good about what each function does and how it behaves, but there is very little about how to get things done, and in what order, and why.

So this is my qualified guess (based on the docs and HelloXR): xrSyncActions sends the pose and when `xrGetActionStatePose` and  `xrLocateSpace` are called the hand and space prediction part is being done and is needed, otherwise the motion compensation api-layer doesn't get the poses.
 
The code I have added is based on the sample HelloXR https://github.com/KhronosGroup/OpenXR-SDK-Source/tree/main/src/tests/hello_xr I have removed the code that shows the controllers as cubes, scales them when you grip and perform haptics on them.

Reply:
I see. I took a look at the code of OpenXR-MotionCompensation layer and it sets its internal state when those functions are called. I think this is clear now, thanks :+1:

Reply:
Would it be a workable alternative for OXRMC to do what it needs itself in `xrEndFrame()` if the actions/action spaces haven't been initialized, or if the controller positions haven't been polled this frame? That would avoid the need for modifications to every game that you want to use with OXRMC that doesn't use controllers

Reply:
Sure, sounds like that would work.